src/main/java/com/synopsys/integration/jenkins/scan/service/scm/SCMRepositoryService.java

```diff
@@ -0,0 +1,78 @@
+package com.synopsys.integration.jenkins.scan.service.scm;
+
+import com.cloudbees.jenkins.plugins.bitbucket.BitbucketSCMSource;
+import com.cloudbees.plugins.credentials.CredentialsProvider;
+import com.cloudbees.plugins.credentials.common.StandardCredentials;
+import com.cloudbees.plugins.credentials.impl.UsernamePasswordCredentialsImpl;
+import hudson.EnvVars;
+import hudson.model.FreeStyleBuild;
+import hudson.model.FreeStyleProject;
+import hudson.model.TaskListener;
+import jenkins.model.Jenkins;
+import jenkins.scm.api.SCMSource;
+import jenkins.scm.api.SCMSourceOwner;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Map;
+
+public class SCMRepositoryService {
+    private final TaskListener listener;
+    private final EnvVars envVars;
+    private static Jenkins jenkins;
+    private static String jobName;
+
+    public SCMRepositoryService(TaskListener listener, EnvVars envVars) {
+        this.listener = listener;
+        this.envVars = envVars;
+    }
+
+    public Object fetchSCMRepositoryDetails(Map<String, Object> scanParameters) {
+        jenkins = Jenkins.getInstanceOrNull();
+        // extracting the job name from the combined job_branch name
+        jobName = envVars.get("JOB_NAME").substring(0, envVars.get("JOB_NAME").indexOf("/"));
+        // getting the pull number from env
+        Integer projectRepositoryPullNumber = envVars.get("CHANGE_ID") != null ? Integer.valueOf(envVars.get("CHANGE_ID")) : 0;
```

Comment:
need to check should we get `0 or null` if `envVars.get("CHANGE_ID") is null`

Reply:
Yes I checked it.  if `envVars.get("CHANGE_ID") is null` then we get 0.

Reply:
I meant, if we pass `null` instead of `0`, will it be a problem?

Reply:
Oh, Let me check.

Reply:
```
{
  "bitbucket": {
    "api": {
      "url": "{URL}",
      "token": "{TOKEN}"
    },
    "project": {
      "key": "BRID",
      "repository": {
        "name": "ninjasworkout",
        "pull": {
          "number": 3
        }
      }
    }
  },
```
What I thought that pull expects an integer that's the reason I passed 0 instead of null.

Reply:
It gives us the following output then we pass null.

```
"project": {
                "repository": {
                    "pull": {},
                    "name": "jenkins-springboot-pipeline-playground"
                },
                "key": "JEN"
            }
```

Reply:
I think It's better for us to put 0 instead of null. As you see it breaks our json structure if we put null.

Reply:
ok got it.

Reply:
**Nitpick**: No need to box a primitive value. Perhaps more efficient one is:
`int projectRepositoryPullNumber = envVars.get("CHANGE_ID") != null ? Integer.parseInt(envVars.get("CHANGE_ID")) : 0;
`